lib/cfn-nag/custom_rules/password_base_rule.rb

```diff
@@ -31,32 +31,34 @@ def audit_impl(cfn_model)
 
     violating_resources.map(&:logical_resource_id)
   end
-end
 
-private
+  private
 
-def verify_parameter_exists(resource, password_property, sub_property_name)
-  if sub_property_name.nil?
-    resource.send(password_property).nil?
-  else
-    resource.send(password_property)[sub_property_name].nil?
+  def property_does_not_exist(resource, password_property, sub_property_name)
+    if resource.send(password_property).nil?
+      true
+    elsif sub_property_name.nil?
+      false
```

Comment:
lol - yes i think this is correct! but the not_exist really turns things upside down